docs/zkapps/o1js/basic-concepts.md

```diff
@@ -27,109 +27,125 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
 
 # o1js Basic Concepts
 
-o1js, fka. SnarkyJS, is a TypeScript (TS) library for writing general-purpose zk programs and writing zk smart contracts for Mina.
+o1js, fka. SnarkyJS, is a TypeScript (TS) library for writing general-purpose ZK programs and writing ZK smart contracts for the Mina Blockchain. In order to create a ZKP (zero knowledge proof), you should use types and operations that can be converted into a ZKP. o1js gives you a lot of different built in types and customizable structures that you can use to create ZKPs.
 
-## Field
+## Field Type
 
-Field elements are the basic unit of data in zero-knowledge proof programming. Each field element can store a number up to almost 256 bits in size. You can think of it as a uint256 in Solidity.
+`Field` elements are the basic unit of data in ZK programming. Every other data type (either built-in or composite) in o1js is made up of `Field` elements.
+
+Each `Field` element can store a number up to almost 256 bits in size. You can think of it as a `uint256` in Solidity, but there are some fundamental differences explained below.
 
 :::note
 
-For the cryptography inclined, the exact max value that a field can store is: 28,948,022,309,329,048,855,892,746,252,171,976,963,363,056,481,941,560,715,954,676,764,349,967,630,336
+For the cryptography inclined, the prime order of the o1js `Field` is: 28,948,022,309,329,048,855,892,746,252,171,976,963,363,056,481,941,560,715,954,676,764,349,967,630,336
 
 :::
 
-For example, in typical programming, you might use:
-
-`const sum = 1 + 3`.
+## Creating a Field Element
 
-In o1js, you write this as:
+You can create a `Field` either with the `new` constructor or by directly calling the `Field` as a constructor. The best practice is to use `Field` without the `new` constructor.
 
-`const sum = new Field(1).add(new Field(3))`
-
-This can be simplified as:
+<!-- prettier-ignore -->
+```ts
+const x = new Field(42);
+const y = Field(923); // Best practice
+```
 
-`const sum = new Field(1).add(3)`
+You can create a `Field` from anything that is "field-like": `number`, `string`, `bigint`, or another `Field`. Note that the argument you give must be an integer in the `Field` range.
 
-Note that the 3 is auto-promoted to a field type to make this cleaner.
+<!-- prettier-ignore -->
+```ts
+const x = Field("12347"); // From string
+const y = Field(172384782343434n); // From bigint
+const z = Field(x); // From another Field
 
-## Built-in data types
+const k = Field("asdf"); // Throws, as this is not a number
+const l = Field("234.34"); // Throws, as this is not an integer
+```
 
-Some common data types you may use are:
+Nevertheless, you can create a `Field` from a negative number. This method implicity calculates the [modular inverse](https://en.wikipedia.org/wiki/Modular_multiplicative_inverse) of the given argument. This is an advanced level information, and you do not need it for simple zkApps.
```

Comment:
This is incorrect. A negative input causes it to create the additive inverse in the field, i.e. takes the number modulo the field size. But "modular inverse" refers to the _multiplicative inverse_ and you also link to an article about multiplicative inverses